ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java

```diff
@@ -189,20 +259,71 @@ private void collectData(File[] replicas, Map<String, BlobStatus> blobIdToStatus
     logger.info("Acceptable Inconsistent blobs count : {}", acceptableInconsistentBlobs);
     logger.info("Inconsistent blobs count due to replication lag : {}", inconsistentDueToReplicationCount);
     logger.info("Real Inconsistent blobs count : {} ", realInconsistentBlobs.size());
-    if (realInconsistentBlobs.size() > 0) {
-      logger.info(
-          "The earliest real inconsistent blob had its last operation at {} ms and difference wrt latest operation time is {} ms",
-          earliestRealInconsistentBlobTimeMs, latestOpTimeMs - earliestRealInconsistentBlobTimeMs);
-    }
     return realInconsistentBlobs;
   }
 
-  private List<String> populateReplicaList(File[] replicas) {
-    List<String> replicaList = new ArrayList<String>();
-    for (File replicaFile : replicas) {
-      replicaList.add(replicaFile.getName());
+  /**
+   * Holds status of a blob from the perspective of an Index. If multiple index entries are found for the same blob,
+   * everything is captured in a single instance of this class
+   */
+  private static class ReplicationStatus {
+    private final Set<File> available = new HashSet<>();
+    private final Set<File> deletedOrExpired = new HashSet<>();
```

Comment:
since you are touching this file, can we separate delete and expired into two separate lists.

Reply:
Don't think it's necessary for what this tool is doing. Delete or expiry means the same to it. Do you see any scenario where making the distinction is useful for this tool? This isn't info that is revealed to the end user.

Reply:
Also opens us up to (unnecessary for this tool) questions like - where does a deleted and expired blob go? We can't repeat entries.

Reply:
fine. I don't see any particular reason why we need it. We can leave it as is for now